static/app/utils/fields/index.ts

```diff
@@ -1575,16 +1583,22 @@ export const getFieldDefinition = (
   }
 };
 
-export function makeTagCollection(fieldKeys: FieldKey[]): TagCollection {
+export function makeTagCollection(
+  fieldKeys: FieldKey[],
+  org: Organization
+): TagCollection {
+  const include_priority = org.features.includes('issue-priority-ui');
   return Object.fromEntries(
-    fieldKeys.map(fieldKey => [
-      fieldKey,
-      {
-        key: fieldKey,
-        name: fieldKey,
-        kind: getFieldDefinition(fieldKey)?.kind,
-      },
-    ])
+    fieldKeys
+      .map(fieldKey => [
+        fieldKey,
+        {
+          key: fieldKey,
+          name: fieldKey,
+          kind: getFieldDefinition(fieldKey)?.kind,
+        },
+      ])
+      .filter(([key, _]) => (key === FieldKey.ISSUE_PRIORITY ? include_priority : true))
```

Comment:
I don't think we actually need to filter it out here, looking at the call site this function won't ever get passed `ISSUE_PRIORITY`